bool grab 
bool index
bool stick
bool x_input
bool y_input
bool stick_down
bool stick_up
